Added hint about timezone in date filters

pkraeutli opened the pull request:
I added a hint about what time zone is used when using date filters.

ahocquard commented:
hello @pkraeutli and thanks for the contribution.

As you are doing a request to a server, it seems logical that it is the timezone of the server that it is used.
Therefore, I don't know if it's useful to precise it in the documentation.

What would be your expectation? Time zone of the user locale connected to the API? Time zone of the code executing the client?

pkraeutli commented:
I don't think anything in the implementation needs to be changed, I just added it because we had some trouble with this and because there is no way to find out the server timezone via the API (at least I don't think so).

In my opinion, it would be best if the client was able to use the same date format for the request as is used in the response (e.g. `product.updated` property), which contains the timezone.

ahocquard commented:
@pkraeutli did you try to pass a date with the ISO-8601 format, as documented here https://api.akeneo.com/documentation/filter.html#on-creation-or-update-date ?

ahocquard commented:
Ok, I did some tests and you're right. 
Problem is only with the filters, and I agree it is not convenient.

Also, the documentation is wrong because it is not ISO-8601 that is accepted in the filters (taht's why I didn't understand at first).
We will try to improve it in the future (but no ETA for now).

ahocquard commented on content/rest-api/filter.md:
```diff
@@ -132,6 +132,7 @@ To get the products that are not in the `camcorders` and `digital_cameras` famil
 
 To filter products on creation or update date, use respectively the product property `created` and `updated`.
 Here are the allowed operators to filter on these properties as well as the corresponding type of value expected in the `search` query parameter.
+Note that dates are interpreted in the time zone of the server that runs Akeneo (e.g. date.timezone setting in php.ini).
```
Could you do that instead please:

```
 :::info
Note that dates are interpreted in the time zone of the server that runs Akeneo (e.g. date.timezone setting in php.ini).
:::
```

pkraeutli commented:
Done 👍

ahocquard commented:
👍  thanks
